tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ManualInstrumentation/Extensions/SpanExtensionsSetTagIntegration.cs

```diff
@@ -0,0 +1,58 @@
+﻿// <copyright file="SpanExtensionsSetTagIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.ManualInstrumentation.Proxies;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Telemetry.Metrics;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.ManualInstrumentation.Extensions;
+
+/// <summary>
+/// Datadog.Trace.ISpan Datadog.Trace.SpanExtensions::SetTag(Datadog.Trace.ISpan,System.String,System.Nullable`1[System.Double]) calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Datadog.Trace.Manual",
+    TypeName = "Datadog.Trace.SpanExtensions",
+    MethodName = "SetTag",
+    ReturnTypeName = "Datadog.Trace.ISpan",
+    ParameterTypeNames = new[] { "Datadog.Trace.ISpan", ClrNames.String, "System.Nullable`1[System.Double]" },
+    MinimumVersion = ManualInstrumentationConstants.MinVersion,
+    MaximumVersion = ManualInstrumentationConstants.MaxVersion,
+    IntegrationName = ManualInstrumentationConstants.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class SpanExtensionsSetTagIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget, TSpan>(ref TSpan span, ref string key, ref double? value)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.SpanExtensions_SetTag);
+
+        // Annoyingly, this takes an ISpan, so we have to do some duckTyping to make it work
+        if (span is IDuckType { Instance: ManualSpan { AutomaticSpan: { } duckTyped } })
+        {
+            // this is the "typical" scenario
+            duckTyped.SetTagInternal(key, value);
+        }
+        else if (span is Span autoSpan)
+        {
+            // Not likely, but technically possible for this to happen
+            autoSpan.SetTagInternal(key, value);
```

Comment:
How exactly would this happen? I can see the first `if` and the `else`, but can't think of how this one would occur

Reply:
Tbh, I can't see how now, but either way, it seems cheap enough that it's not worth worrying about too much 😄 

Btw, this implementation has changed a lot in #5214 😬